### PR TITLE
🐞 fix(utils): run the model in eval() so the weights of the model itself will not be modified

### LINKED ISSUE #2965: 🐞 CS-Flow Model initializes EfficientNet-B5 with NaN BatchNorm running_var
### Describe the bug

When I initialize the CSFLowModel and try to train, I always got NaN values returned after forwarding. With this NaN values in the first train epoch it is not possible to calucalte the loss values and thus the whole training fails.

 I found out that this comes from some layers from the FeatureExtraction which have NaNs inside:

```python
import torch
import torchvision
import sys
import platform
from anomalib.models.image.csflow.torch_model import CsFlowModel

info = {
    "python": sys.version.replace("\n", " "),
    "platform": platform.platform(),
    "torch": torch.__version__,
    "torchvision": torchvision.__version__,
    "cuda_available": str(torch.cuda.is_available()),
    "cuda_version": torch.version.cuda or "None",
    "cudnn_version": str(torch.backends.cudnn.version()) if torch.backends.cudnn.is_available() else "None",
    "gpu_name": torch.cuda.get_device_name(0) if torch.cuda.is_available() else "None",
    "compute_capability": str(torch.cuda.get_device_capability(0)) if torch.cuda.is_available() else "None",
    "allow_tf32_matmul": str(getattr(torch.backends.cuda.matmul, "allow_tf32", None)),
    "allow_tf32_cudnn": str(getattr(torch.backends.cudnn, "allow_tf32", None)),
}
for key, value in info.items():
    print(key, value)

model = CsFlowModel(
    input_size=(640, 1280),
    cross_conv_hidden_channels=1024,
    n_coupling_blocks=4,
    clamp=3,
    num_channels=3,
).eval()

print("\n# Probe BN in backbone: features.0.1")
bn = model.feature_extractor.feature_extractor.backbone.features[0][1]  # BatchNorm2d
def s(t): return (torch.isfinite(t).all().item(),
                  float(t.min().item()), float(t.max().item()), float(t.mean().item()))
print("affine weight finite:", torch.isfinite(bn.weight).all().item())
print("affine bias   finite:", torch.isfinite(bn.bias).all().item())
print("running_mean  stats :", s(bn.running_mean))
print("running_var   stats :", s(bn.running_var))
```

My output:
```
# Probe BN in backbone: features.0.1
affine weight finite: True
affine bias   finite: True
running_mean  stats : (False, nan, nan, nan)
running_var   stats : (False, nan, nan, nan)
```

Can anyone confirm this? Do I miss something?

### Dataset

N/A

### Model

CSFlow

### Steps to reproduce the behavior

see script above

### OS information

python 3.10.12 (main, Aug 15 2025, 14:32:43) [GCC 11.4.0]
platform Linux-5.15.148-tegra-aarch64-with-glibc2.35
torch 2.8.0
torchvision 0.23.0
cuda_available True
cuda_version 12.6
cudnn_version 90300
gpu_name Orin
compute_capability (8, 7)
allow_tf32_matmul False
allow_tf32_cudnn True

### Expected behavior

BatchNorm buffers (`running_mean`, `running_var`) of EfficientNet-B5 are finite right after `CsFlowModel` construction.

### Screenshots

_No response_

### Pip/GitHub

pip

### What version/branch did you use?

v2.1

### Configuration YAML

```yaml
no configuration yaml file needed
```

### Logs

```shell
see above
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by monzelr**
## 📝 Description

- During the build up process of the FeatureExtraction model, the FeatureExtraction model runs in training modes and modifies its weights. The weihgts of the FeatureExtraction Model shall never be modified.
- 🛠️ Fixes #2965

## ✨ Changes

Select what type of change your PR is:
- [x ] 🐞 Bug fix (non-breaking change which fixes an issue)

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).
- [x ] 🏷️ My PR title follows conventional commit format.

For more information about code review checklists, see the [Code Review Checklist](https://github.com/open-edge-platform/anomalib/blob/main/docs/source/markdown/guides/developer/contributing.md).

### CODE REVIEW

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks for fix

**Review by rajeshgangireddy — APPROVED** [review]
Thanks for spotting the issue and fixing it
